# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/msaa/EnumVariant.h
# Commit: 1a948c0ec982
# Full Hash: 1a948c0ec9826d73bba860699f540cbb25f34e34
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-04-29 21:42:31
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 18: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 3ea7b46bb4f8 -r 1a948c0ec982 accessible/windows/msaa/EnumVariant.h
--- a/accessible/windows/msaa/EnumVariant.h	Thu Apr 29 10:20:56 2021 +0000
+++ b/accessible/windows/msaa/EnumVariant.h	Thu Apr 29 10:20:57 2021 +0000
@@ -7,8 +7,8 @@
 #ifndef mozilla_a11y_EnumVariant_h__
 #define mozilla_a11y_EnumVariant_h__
 
-#include "AccessibleWrap.h"
 #include "IUnknownImpl.h"
+#include "MsaaAccessible.h"
 
 namespace mozilla {
 namespace a11y {
@@ -18,9 +18,9 @@
  */
 class ChildrenEnumVariant final : public IEnumVARIANT {
  public:
-  explicit ChildrenEnumVariant(AccessibleWrap* aAnchor)
-      : mAnchorAcc(aAnchor),
-        mCurAcc(mAnchorAcc->LocalChildAt(0)),
+  explicit ChildrenEnumVariant(MsaaAccessible* aAnchor)
+      : mAnchorMsaa(aAnchor),
+        mCurAcc(mAnchorMsaa->LocalAcc()->LocalChildAt(0)),
         mCurIndex(0) {}
 
   // IUnknown
@@ -45,13 +45,13 @@
   ChildrenEnumVariant& operator=(const ChildrenEnumVariant&) = delete;
 
   ChildrenEnumVariant(const ChildrenEnumVariant& aEnumVariant)
-      : mAnchorAcc(aEnumVariant.mAnchorAcc),
+      : mAnchorMsaa(aEnumVariant.mAnchorMsaa),
         mCurAcc(aEnumVariant.mCurAcc),
         mCurIndex(aEnumVariant.mCurIndex) {}
   virtual ~ChildrenEnumVariant() {}
 
  protected:
-  RefPtr<AccessibleWrap> mAnchorAcc;
+  RefPtr<MsaaAccessible> mAnchorMsaa;
   LocalAccessible* mCurAcc;
   uint32_t mCurIndex;
 };